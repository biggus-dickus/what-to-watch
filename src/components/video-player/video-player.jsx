import React from 'react';
import PropTypes from 'prop-types';


export default class VideoPlayer extends React.PureComponent {
  constructor(props) {
    super(props);

    this.ref = React.createRef();

    this.state = {
      isPlaying: false
    };
  }

  render() {
    return <video
      ref={this.ref}
      {...this.props}
      onMouseEnter={this._handleMouseEnter}
      onMouseLeave={this._handleMouseLeave} />;
  }

  componentDidUpdate(prevProps, prevState) {
    if (!prevState.isPlaying && this.state.isPlaying) {
      this.ref.current.play();
    }

    if (prevState.isPlaying && !this.state.isPlaying) {
      this.ref.current.currentTime = 0;
    }
  }

  _handleMouseEnter = () => {
    // setTimeout(() => this.setState({isPlaying: true}), 1000);
    this.setState({isPlaying: true})
  };

  _handleMouseLeave = () => this.setState({isPlaying: false});
}


VideoPlayer.propTypes = {
  muted: PropTypes.bool,
  poster: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired
};
