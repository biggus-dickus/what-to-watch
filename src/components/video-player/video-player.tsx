import * as React from 'react';

import {MEDIA_INITIAL_VOLUME} from '../../config/config';
import {secToHms} from '../../utilities/helpers';

interface Props {
  show: boolean,
  duration: number,
  filmName: string,
  onClose: () => void,
  poster: string,
  src: string
}

interface State {
  readonly isPlaying: boolean,
  readonly isPaused: boolean,
  readonly timeElapsed: number,
  readonly timeElapsedPercentage: number,
}

const togglePreventScroll = (toggle) => {
  if (toggle) {
    document.body.classList.add(`no-scroll`);
    return;
  }

  document.body.classList.remove(`no-scroll`);
};

const TWO_DIGIT_INT = 10;
const prependZero = (val: number): string => (val < TWO_DIGIT_INT) ? `0${val}` : `${val}`;


export default class VideoPlayer extends React.PureComponent<Props, State> {
  private _ref = React.createRef<HTMLVideoElement>();
  private _interval: NodeJS.Timer;

  constructor(props) {
    super(props);

    this.state = {
      isPlaying: false,
      isPaused: false,
      timeElapsed: this.timeElapsed,
      timeElapsedPercentage: this.timeElapsedPercentage
    };
  }

  get videoDuration(): number {
    if (this._ref.current && this._ref.current.duration) {
      return this._ref.current.duration;
    }

    // Convert runTime in minutes to seconds
    return this.props.duration * 60;
  }

  get timeElapsed(): number {
    if (this._ref.current && this._ref.current.currentTime) {
      return this._ref.current.currentTime;
    }

    return 0;
  }

  get timeElapsedPercentage(): number {
    const {current} = this._ref;
    if (current && current.currentTime && current.duration) {
      return current.currentTime / current.duration * 100;
    }

    return 0;
  }

  componentDidMount(): void {
    window.addEventListener(`keyup`, this._onEscPress);
  }

  componentDidUpdate(prevProps: Props, prevState: State): void {
    // Prevent scroll
    if (!prevProps.show && this.props.show) {
      togglePreventScroll(true);
    }

    // Allow scroll
    if (prevProps.show && !this.props.show) {
      togglePreventScroll(false);
    }

    if (this._ref.current) {
      // Start playback
      if (!prevState.isPlaying && this.state.isPlaying) {
        this._ref.current.play();
      }

      // Pause playback
      if (!prevState.isPaused && this.state.isPaused) {
        this._ref.current.pause();
      }
    }
  }

  componentWillUnmount(): void {
    window.removeEventListener(`keyup`, this._onEscPress);
  }

  render() {
    const {show, filmName, poster, src} = this.props;
    const {isPaused, isPlaying, timeElapsed, timeElapsedPercentage} = this.state;

    if (show) {
      let playBtnIcon = (
        <svg viewBox="0 0 19 19" width="19" height="19" data-test="at-play-btn-icon">
          <use xlinkHref="img/sprite/sprite.svg#play-s" />
        </svg>
      );

      let playBtnText = `Play`;

      const classList = [`player`];
      if (isPlaying) {
        classList.push(`is-playing`);
        if (classList.includes(`is-paused`)) {
          classList.splice(classList.indexOf(`is-paused`), 1);
        }
      }

      if (isPaused) {
        classList.push(`is-paused`);
        if (classList.includes(`is-playing`)) {
          classList.splice(classList.indexOf(`is-playing`), 1);
        }

        playBtnIcon = (
          <svg viewBox="0 0 14 21" width="14" height="21" data-test="at-pause-btn-icon">
            <use xlinkHref="img/sprite/sprite.svg#pause" />
          </svg>
        );

        playBtnText = `Pause`;
      }

      const [h, m, s] = secToHms(this.videoDuration);

      return (
        <div className={classList.join(` `)}>
          <video
            className="player__video"
            ref={this._ref}
            // onPlay={this.onPlay}
            // onPause={this.onPause}
            {...{poster, src}} />

          <button type="button" className="player__exit" onClick={this._onExit}>
            Exit
          </button>

          <div className="player__controls">
            <div className="player__controls-row">
              <div className="player__time">
                <progress
                  className="player__progress"
                  value={timeElapsed}
                  max={this.videoDuration}
                  onClick={this._setCurrentTime} />
                <button
                  type="button"
                  className="player__toggler"
                  style={{left: `${timeElapsedPercentage.toFixed(2)}%`}}>
                  Toggler
                </button>
              </div>

              <div className="player__time-value">
                {prependZero(h)}:{prependZero(m)}:{prependZero(s)}
              </div>
            </div>

            <div className="player__controls-row">
              <button
                type="button"
                className="player__play"
                data-test="at-play-btn"
                onClick={this._handlePlaybackToggle}>
                {playBtnIcon}
                <span>{playBtnText}</span>
              </button>

              <div className="player__name">{filmName}</div>

              <button
                type="button"
                className="player__full-screen"
                onClick={this._handleRequestFullScreen}>
                <svg viewBox="0 0 27 27" width="27" height="27">
                  <use xlinkHref="img/sprite/sprite.svg#full-screen" />
                </svg>
                <span>Full screen</span>
              </button>

            </div>
          </div>
        </div>
      );
    }

    return null;
  }

  // onPlay = () => console.log('play')
  // onPause = () => console.log('pause')

  _setCurrentTime = (e): void => {
    if (this._ref.current && this.state.timeElapsed) {
      const rect = e.target.getBoundingClientRect();
      const clickSpot = e.clientX - rect.left;
      const percentage = (clickSpot / rect.width).toFixed(2);

      this._ref.current.currentTime = this.videoDuration * +percentage;
    }
  };

  _setTimings = (): void => {
    this._interval = setInterval(() => {
      this.setState({
        timeElapsed: this.timeElapsed,
        timeElapsedPercentage: this.timeElapsedPercentage
      });
    }, 1000);
  };

  _freezeTimings = (): void => {
    clearInterval(this._interval);
  };

  _clearTimings = (): void => {
    clearInterval(this._interval);
    this.setState({
      timeElapsed: 0,
      timeElapsedPercentage: 0
    });
  };

  _handleRequestFullScreen = (): void => {
    if (this._ref.current.requestFullscreen) {
      this._ref.current.requestFullscreen();
    }
  };

  _handlePlaybackToggle = ():void => {
    if (!this.state.isPlaying && !this.state.isPaused) {
      this._ref.current.volume = MEDIA_INITIAL_VOLUME;

      this._setTimings();
      this.setState({isPlaying: true});
      return;
    }

    if (this.state.isPlaying) {
      this._freezeTimings();
    } else if (this.state.isPaused) {
      this._setTimings();
    }

    this.setState((state) => ({
      isPlaying: !state.isPlaying,
      isPaused: !state.isPaused
    }));
  };

  _onExit = () => {
    if (this.state.isPlaying || this.state.isPaused) {
      this._ref.current.load();
      this._clearTimings();
      this.setState({isPlaying: false, isPaused: false});
    }

    this.props.onClose();
  };

  _onEscPress = (e: KeyboardEvent) => {
    if (e.key === `Escape` && this.props.show) {
      this._onExit();
    }
  };
}
