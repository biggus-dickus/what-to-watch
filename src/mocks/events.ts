// JSDom has no implementation for common methods of HTML media elements.
// Tests will fail without this. Sad but true.
// https://github.com/jsdom/jsdom/issues/2155#issuecomment-366703395
export const mockHTMLMediaMethods = (): void => {
  window[`HTMLMediaElement`].prototype.load = () => { /* do nothing */ };
  window[`HTMLMediaElement`].prototype.play = () => { /* do nothing */ };
  window[`HTMLMediaElement`].prototype.pause = () => { /* do nothing */ };
};
