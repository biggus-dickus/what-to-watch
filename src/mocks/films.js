export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picUrl: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Bohemian Rhapsody`,
    picUrl: `img/bohemian-rhapsody.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Macbeth`,
    picUrl: `img/macbeth.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Aviator`,
    picUrl: `img/aviator.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `We need to talk about Kevin`,
    picUrl: `img/we-need-to-talk-about-kevin.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `What We Do in the Shadows`,
    picUrl: `img/what-we-do-in-the-shadows.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Revenant`,
    picUrl: `img/revenant.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Johny English`,
    picUrl: `img/johnny-english.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Shutter Island`,
    picUrl: `img/shutter-island.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Pulp Fiction`,
    picUrl: `img/pulp-fiction.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `No Country for Old Men`,
    picUrl: `img/no-country-for-old-men.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Snatch`,
    picUrl: `img/snatch.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Moonrise Kingdom`,
    picUrl: `img/moonrise-kingdom.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Seven Years in Tibet`,
    picUrl: `img/seven-years-in-tibet.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Midnight Special`,
    picUrl: `img/midnight-special.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `War of the Worlds`,
    picUrl: `img/war-of-the-worlds.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Dardjeeling Limited`,
    picUrl: `img/dardjeeling-limited.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Orlando`,
    picUrl: `img/orlando.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Mindhunter`,
    picUrl: `img/mindhunter.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
];
