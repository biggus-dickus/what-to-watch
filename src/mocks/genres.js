export const Genre = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};
