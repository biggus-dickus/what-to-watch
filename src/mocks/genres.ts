import {Genre as GenreType} from '../types'; // eslint-disable-line

export const Genre = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};

export const mockGenres = Object.values(Genre);

export const mockGenreEntity: GenreType = {
  currentGenre: Genre.CRIME,
  genres: [...mockGenres],
  onGenreChange: jest.fn()
};
