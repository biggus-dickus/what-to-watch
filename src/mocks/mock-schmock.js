export const mockGenres = [`All`, `BDSM`, `Gay gymnastics`, `Psycho-dramas for aestheticizing faggots`];

export const mockFilms = [
  {
    name: `L'elephant vert`,
    genre: [mockGenres[1], mockGenres[2]],
    pageUrl: `https://www.imdb.com/title/tt1003080/`,
    picUrl: `http://lurkmore.so/images/4/47/Slonik_macro.jpg`,
    videoUrl: `https://storage.googleapis.com/coverr-main/mp4%2FCam-Man.mp4`
  },
  {
    name: `Sal√≤ o le 120 giornate di Sodoma`,
    genre: [mockGenres[3]],
    pageUrl: `https://www.imdb.com/title/tt0073650/`,
    picUrl: `https://m.media-amazon.com/images/M/MV5BYjE0NTU3MTUtYjE0OC00ZGM1LWIwNzYtNDU2YWI2YWJjZjEwXkEyXkFqcGdeQXVyMTA2ODMzMDU@._V1_SY1000_CR0,0,764,1000_AL_.jpg`,
    videoUrl: `etc/na-zdorovnya.mp4`
  },
  {
    name: `Un Chien Andalou`,
    genre: [mockGenres[3]],
    pageUrl: `https://www.imdb.com/title/tt0020530/`,
    picUrl: `https://m.media-amazon.com/images/M/MV5BMjZlY2RjM2ItZGEwOC00ODQwLWFhZDYtMjE3ODMyYWNiOWNlXkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SY1000_CR0,0,706,1000_AL_.jpg`,
    videoUrl: `etc/na-zdorovnya.mp4`
  }
];
