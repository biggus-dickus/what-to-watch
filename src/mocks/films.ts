import {Film} from '../types'; // eslint-disable-line
import {Genre} from './genres';

export const mockFilms: Film[] = [
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    id: 1,
    genre: Genre.FAMILY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    bgColor: `green`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 7.77,
    released: 2019,
    runTime: 120,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Bohemian Rhapsody`,
    id: 2,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    bgColor: `blue`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1654,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Macbeth`,
    id: 3,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/macbeth.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4%2FCam-Man.mp4`
  },
  {
    bgColor: `rebeccapurple`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 5.55,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Aviator`,
    id: 4,
    genre: Genre.ROMANCE,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/aviator.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/5/5b/Marnie_%281964%29_trailer.webm/Marnie_%281964%29_trailer.webm.360p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `We need to talk about Kevin`,
    id: 5,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/5/5b/Marnie_%281964%29_trailer.webm/Marnie_%281964%29_trailer.webm.360p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `What We Do in the Shadows`,
    id: 6,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/what-we-do-in-the-shadows.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/2/23/Dr_Strangelove_-_Official_Trailer.webm/Dr_Strangelove_-_Official_Trailer.webm.480p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Revenant`,
    id: 7,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/revenant.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Harvesting%20Rice.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 8.10,
    released: 2005,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Johny English`,
    id: 8,
    genre: Genre.COMEDIES,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/johnny-english.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Lake%20Day.mp4`
  },
  {
    bgColor: `pink`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 2009,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Shutter Island`,
    id: 9,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/shutter-island.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Deserted%20Island.mp4`
  },
  {
    bgColor: `purple`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Quentin Tarantino`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 9.50,
    released: 1994,
    runTime: 180,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Pulp Fiction`,
    id: 10,
    genre: Genre.CRIME,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/pulp-fiction.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Stunning%20Mountains.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `No Country for Old Men`,
    id: 11,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/no-country-for-old-men.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Gubat%20Beach.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Guy Ritchie`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 8.36,
    released: 2004,
    runTime: 125,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Snatch`,
    id: 12,
    genre: Genre.COMEDIES,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/snatch.jpg`,
    previewVideo: `etc/na-zdorovnya.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Moonrise Kingdom`,
    id: 13,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/moonrise-kingdom.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Caramoan%20Drone.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Seven Years in Tibet`,
    id: 14,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/seven-years-in-tibet.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Ants%20Walking.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Midnight Special`,
    id: 15,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/midnight-special.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Boat%20in%20Caramoan.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `War of the Worlds`,
    id: 16,
    genre: Genre.SCI_FI,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/war-of-the-worlds.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Blue%20Joy.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Dardjeeling Limited`,
    id: 17,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/dardjeeling-limited.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Stop-Sign.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Orlando`,
    id: 18,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/orlando.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Steaming-Yellowstone.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Mindhunter`,
    id: 19,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/mindhunter.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/California-Desert.mp4`
  }
];
