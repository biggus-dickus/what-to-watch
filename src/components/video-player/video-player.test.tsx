import * as React from 'react';
import * as renderer from 'react-test-renderer';
import * as Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';

import {mockFilms} from '../../mocks/films';
import {PlayState} from './video-player'; // eslint-disable-line

import VideoPlayer from './video-player';

configure({adapter: new Adapter()});


const props = {
  show: true,
  duration: mockFilms[1].runTime,
  filmName: mockFilms[1].name,
  onClose: jest.fn(),
  poster: mockFilms[1].posterImage,
  src: mockFilms[1].video
};


describe(`VideoPlayer test suite`, () => {
  it(`<VideoPlayer /> renders correctly`, () => {
    const tree = renderer.create(<VideoPlayer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should not render anything when the "show" prop is false`, () => {
    const newProps = {...props, show: false};
    const wrapper = shallow(<VideoPlayer {...newProps} />);

    expect(wrapper.find(`video`)).toHaveLength(0);
  });

  it(`should correctly toggle the "Play" button when paused`, () => {
    const wrapper = shallow(<VideoPlayer {...props} />);

    expect(wrapper.find(`[data-test="at-play-btn"]`).text()).toEqual(`Play`);
    expect(wrapper.find(`[data-test="at-play-btn-icon"]`)).toHaveLength(1);
    expect(wrapper.find(`[data-test="at-pause-btn-icon"]`)).toHaveLength(0);

    wrapper.setState({play: PlayState.PLAY});

    expect(wrapper.find(`[data-test="at-play-btn"]`).text()).toEqual(`Pause`);
    expect(wrapper.find(`[data-test="at-pause-btn-icon"]`)).toHaveLength(1);
    expect(wrapper.find(`[data-test="at-play-btn-icon"]`)).toHaveLength(0);
  });
});
