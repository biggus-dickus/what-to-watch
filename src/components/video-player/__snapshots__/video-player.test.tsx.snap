// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer test suite <VideoPlayer /> renders correctly 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    onPause={[Function]}
    onPlay={[Function]}
    poster="poster.jpg"
    src="https://thepiratebay.org"
  />
  <button
    className="player__exit"
    data-test="at-exit-btn"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max={7200}
          onClick={[Function]}
          value={0}
        />
        <button
          className="player__toggler"
          style={
            Object {
              "left": "0.00%",
            }
          }
          type="button"
        >
          Sliding thumb
        </button>
      </div>
      <div
        className="player__time-value"
      >
        02
        :
        00
        :
        00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        data-test="at-play-btn"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-test="at-play-btn-icon"
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="img/sprite/sprite.svg#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Bohemian Rhapsody
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="img/sprite/sprite.svg#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
